### DISCUSSION TITLE
how to use C # to call fastdds

### DISCUSSION THREAD

**Discussion by huajuhong** [eProsima/Fast-DDS#4637] [Q&A]
Hello everyone, how to use C # to call fastdds

**Comment by JesusPoderoso** [comment]
Hi @Hsiaye, I am afraid there is no C# wrapper for Fast DDS.
Sorry for the inconvenience.

**Reply by huajuhong** [reply]
Thank you for your reply, I would like to discuss with you that it is currently being used on the Windows platform and developed in C# language. If it is possible to export it through P/Invoke or generate a DLL using COM (compliant with clr), then it can be called by C#

**Reply by JesusPoderoso** [reply]
After a quick research, It looks like P/Invoke will not meet your needs. I would recommend you to use [SWIG](https://www.swig.org/), by creating a simple API that uses Fast DDS, and exporting that API to C# using SWIG wrapper.
If you need more help in relation to that, do not hesitate contacting our commercial support: support@eprosima.com
